pkg/k8s/scanner/io.go

```diff
@@ -15,29 +16,43 @@ import (
 
 var r = regexp.MustCompile("\\\\|/|:|\\*|\\?|<|>")
 
-func createTempFile(artifact *artifacts.Artifact) (string, error) {
-	filename := fmt.Sprintf("%s-%s-%s-*.yaml", artifact.Namespace, artifact.Kind, artifact.Name)
-
-	if runtime.GOOS == "windows" {
-		// removes characters not permitted in file/directory names on Windows
-		filename = filenameWindowsFriendly(filename)
-	}
-	file, err := os.CreateTemp("", filename)
+// generateTempFolder creates a folder with yaml files generated from kubernetes artifacts
+// returns a folder name, a map for mapping a temp target file to k8s artifact and error
+func generateTempFolder(arts []*artifacts.Artifact) (string, map[string]*artifacts.Artifact, error) {
+	tempFolder, err := os.MkdirTemp("", "trivyk8s*")
 	if err != nil {
-		return "", xerrors.Errorf("creating tmp file error: %w", err)
+		return "", nil, xerrors.Errorf("failed to create temp folder: %w", err)
 	}
-	defer func() {
+
+	m := make(map[string]*artifacts.Artifact)
+	for _, artifact := range arts {
+		filename := fmt.Sprintf("%s-%s-%s-*.yaml", artifact.Namespace, artifact.Kind, artifact.Name)
```

Comment:
What if we put the loop body in a separate function for readability? Then the error will only need to be logged once

Reply:
done https://github.com/aquasecurity/trivy/pull/7690/commits/c941d62f3a84567117268e95dbd9d7a9a42f140f